ddtrace/tracer/textmap.go

```diff
@@ -591,7 +591,7 @@ func overrideDatadogParentID(ctx, w3cCtx, ddCtx *spanContext) {
 	ctx.spanID = w3cCtx.spanID
 	if w3cCtx.reparentID != "" {
 		ctx.reparentID = w3cCtx.reparentID
-	} else if ddCtx != nil {
```

Comment:
note: technically we _do_ return early if `ddCtx` is `nil`, so this check wouldn't be necessary, but I wonder if there could be an instance in which the value is changed between the check and this line. From tracing the code, it seems that this function is only called from one location that doesn't lock. I'm open to reverting this and keeping the check, especially if we anticipate changing code to cause concurrency here.